### DISCUSSION TITLE
Generating Dataset for instability prediction using machine learning

### DISCUSSION THREAD

**Discussion by Izzy-Elizzy** [google-deepmind/torax#317] [Ideas]
Hello TORAX team,
I'm working on a project about plasma instabilities in tokamaks, focusing on tearing modes. I'm looking to use machine learning to predict these instabilities. Since actual experimental data is very hard to access, I'm wondering if TORAX can help generate synthetic datasets for training my model. Specifically:

Can TORAX simulate scenarios that might lead to tearing instabilities?
What key parameters from these simulations would be most relevant?
Is it possible to label the outputs as stable or potentially unstable?

Any guidance on setting up TORAX for this purpose would be greatly appreciated. I'm also considering using quantum machine learning techniques, so any thoughts on adapting TORAX outputs for this would be helpful.
Thanks for your time!

**Comment by jcitrin** [comment]
Apologies for the delayed reply and thanks for your interest!

> Can TORAX simulate scenarios that might lead to tearing instabilities?

In principle yes, TORAX can simulate scenarios which are susceptible to tearing instabilities.

> What key parameters from these simulations would be most relevant?

Things like pressure gradients, magnetic field, q-profile. See e.g:
https://iopscience.iop.org/article/10.1088/1741-4326/ad0488

> Is it possible to label the outputs as stable or potentially unstable?

Not at the moment. You need information from additional modules that observe dynamic TORAX outputs. e.g., something like: https://www.osti.gov/pages/servlets/purl/2007206 . It is very difficult to predict this from theory + modelling since NTM triggering is a stochastic nonlinear event, requiring a seed island from a perturbation like sawteeth, RMPs. Therefore you'd need a data-driven method as above, or at least some rough heuristics like q_min>1, beta and/or beta' below some limit, etc., and then see if scenarios can be maintained away from these limits. 

> I'm also considering using quantum machine learning techniques, so any thoughts on adapting TORAX outputs for this would be helpful.

I'm not sure exactly what you mean here, so it is difficult to answer.

In general, in the near future, we plan on significantly extending the amount of TORAX outputs, e.g. including 0D quantities and more magnetic geometry quantities. This may be helpful for you.

Hope this helps!
